### DISCUSSION TITLE
TST 3.9.22 fails to keep/save tree structure in Firefox 123.1

### DISCUSSION THREAD

**Discussion by dschadt** [piroor/treestyletab#3469] [General]
the last signed and official version is TST 3.9.22. Since updateing FF to v123.1 the tree structure is flat and all structure is lost.
Updating to TST 4.01 fails in FF as it is not signed and therfore in FF standard not allowed.
xpinstall.signatures.required = false has no effect

**Comment by irvinm** [comment]
I'm running TST 3.9.22 and FF 123.0.1 and have no problems.  Have you tried a new profile just as a test to see if that works for you?

You can't use an unsigned version in FF proper.  You can only use it in the Dev Edition or Nightly with the about:config change.

**Reply by irvinm** [reply]
@dschadt can you try 4.0.3?  It is the main release now.

**Reply by dschadt** [reply]
no change, any tree branch will reset to flat structure after restart of FF

**Reply by dschadt** [reply]
with new profile you mean new FF profile? Need to safe the tabs first. There is no easy backup&restore as with a new FF profile the favorites will go away too.

**Reply by irvinm** [reply]
For a new profile:  Tabs, you can use the addon "Tab Session Manager" to save\export\import tab sessions ... and it supports TST.  For favorites, as long as you use Firefox Sync, those will be restored in a new profile. 

I'm not saying you have to stick with a new profile, but to just try things out to see if it works.  If it works as expected, you can then try to determine what is wrong\different with your existing profile *OR* try to move everything to your new profile.  Think of it as just a test for now.

**Reply by irvinm** [reply]
If there was a broad bug around losing tree structures on restart, there would be a ton of conversation ongoing about that.  It has to be something in your profile or environment that is causing this behavior.
